### DISCUSSION TITLE
Configure Valheim Plus

### DISCUSSION THREAD

**Discussion by Grafal436** [mbround18/valheim-docker#939] [Q&A] [answered]
Hallo everyone,

I'm trying to get Valheim Plus to work on my Docker setup, but I'm running into some issues. I'm following the instructions [here](https://github.com/mbround18/valheim-docker/blob/main/docs/tutorials/getting_started_with_mods.md). 

According to the documentation, the Docker container should have Valheim Plus and BepInEx installed already with
`TYPE=ValheimPlus`

I tried to connect to the container with 
`docker exec -it valheim_valheim_1 /bin/bash`
which works fine
However, I can't find any trace of "BepInEx", or the "plugins" and "config" directories. I also can't find the valheim_plus.cfg file. The only folder I can locate is the steamapps folder, but there's no common/Valheim/BepInEx/ or similar structure. (According to Windows)

What did I do wrong?
Is there something more I need to configure?
How can I verify if Valheim Plus is working?
Where is the valheim_plus.cfg file supposed to be located?

Greetings,

config:

```
> version: "3"
> services:
>   valheim:
>     image: mbround18/valheim:latest
>     stop_signal: SIGINT
>     ports:
>       - "2456:2456/udp"
>       - "2457:2457/udp"
>       - "2458:2458/udp"
>     environment:
>       PORT: 2456
>       NAME: "XYZ"
>       WORLD: "Dedicated"
>       PASSWORD: "XYZ"
>       TZ: "Austria/Vienna"
>       PUBLIC: 1
>       AUTO_BACKUP: 1
>       AUTO_BACKUP_SCHEDULE: "0 2 * * *"
>       AUTO_BACKUP_DAYS_TO_LIVE: 6
>       AUTO_BACKUP_ON_UPDATE: 1
>       AUTO_BACKUP_ON_SHUTDOWN: 0
>       UPDATE_ON_STARTUP: 0
>       TYPE: "ValheimPlus"
> 
>     volumes:
>     - /root/home/valheim/saves:/root/home/steam/.config/unity3d/IronGate/Valheim
>     - /root/home/valheim/server:/root/home/steam/valheim
>     - /root/home/valheim/backups:/root/home/steam/backups
```

**Comment by mbround18** [comment, marked as answer]
The option ValheimPlus was deprecated and you should have gotten a message. basically Valheim plus stopped maintaining their plugin.

it hasnt been updated since 2022 https://github.com/valheimPlus/ValheimPlus/releases

**Reply by Grafal436** [reply]
Now that makes sense.
I just tested it with TYPE: BepInEx
and it works just fine.

Thanks

**Reply by andrewdunndev** [reply]
Bumping the discussion since I looked at this today. I'm coming from [this project](https://github.com/lloesche/valheim-server-docker) to find a container that is maintained. I am not super comfortable maintaining a "mod pack" for my server so ValheimPlus really works for me being an "all in one" mod for what I'd like. As @mbround18 noted above the upstream project is abandoned, however there seems to be an active fork [over here](https://github.com/Grantapher/ValheimPlus/) with recent releases. The previous container I was using was referencing that fork.

To get this all working in this projects container I did the following:

```
TYPE=BepInEx
MODS=https://github.com/Grantapher/ValheimPlus/releases/download/0.9.14.0/UnixServer.zip
```

This worked for me, I was able to bring my V+ configuration over as well as my world saves.

@mbround18 would it be useful to:
 - option 1) remove ValheimPLus as a `TYPE` from this container?
 - option 2) transition to the fork I linked above and keep it available for people?

**Reply by mbround18** [reply]
Hmm it looks like that mod hasn't been updated in over a month, when I get home I can look at it :)

**Reply by andrewdunndev** [reply]
No rush, I just noticed that there may be some users funneling your way due to this [this discussion](https://github.com/lloesche/valheim-server-docker/issues/648). There is an active error in the other project that causes the server to hang after executing an update. Likely more people impacted by this will stumble into your project and wonder "how do I V+". 

I approached the "use BepInEx" idea with gusto but I quickly became intimidated by how hard it would be to replicated what V+ is doing by assembling a large mod pack. Since I am sort of subverting your container to accomplish what other users may also wanted to do I felt it was a good idea to create a breadcrumb for folks.

It's a shame that the V+ thing became such a mess.

**Reply by mbround18** [reply]
I agree, tbh am looking to make that simplier. btw heres the patch for those docs https://github.com/mbround18/valheim-docker/pull/969

I am aiming to add support for R2MODMAN_MOD_LIST in the future where you can just copy the modpack list 

![image](https://github.com/user-attachments/assets/6391ae88-c0a6-413b-91b3-b31c4943b184)

but thats something im still kicking around as a potential, if youd like that feature let me know!

**Comment by shad-lp** [comment]
I am in the exact same case as @andrewgdunn
It would be lovely to get that option back !

**Reply by mbround18** [reply]
@shad-lp and @andrewgdunn for ValheimPlus installation use: 

```
    environment:
      - TYPE=BepInEx
      - MODS=
        https://github.com/Grantapher/ValheimPlus/releases/download/0.9.15.0/ValheimPlus.dll
```

I can add the option back to automate this in the future but for now this should get it installed and goin
